File: doc/codes/csharp/examples/api/compute/pattern/laguerre-gaussian-example.cs
using System.Numerics;
using AUTD3;
using static AUTD3.Units;

// HIDE
namespace DocSamples.ApiComputePatternLaguerreGaussianExample;

internal static class Sample
{
    internal static void Run()
    {
        // HIDE_END
var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });

var phases = geometry.PhaseBuffer();
var intensities = geometry.IntensityBuffer();

var target = geometry.Center + new Vector3(0.0f, 0.0f, 150.0f);
var option = new LaguerreGaussianOption(p: 0, l: 1, waist: 10.0f * mm);
var wavelength = Pattern.Wavelength(340.0f * m / s);
Pattern.LaguerreGaussianPhase(geometry, target, Vector3.UnitZ, option, wavelength, phases);
Pattern.LaguerreGaussianIntensity(geometry, target, Vector3.UnitZ, option, wavelength, intensities);
        // HIDE
    }
}
// HIDE_END

File: doc/codes/python/examples/api/compute/pattern/laguerre-gaussian-example.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import m, s
from autd3_pattern import (
    LaguerreGaussianOption,
    laguerre_gaussian_phase,
    laguerre_gaussian_intensity,
    wavelength,
)

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

phases = geometry.phase_buffer()
intensities = geometry.intensity_buffer()

target = geometry.center() + np.array([0.0, 0.0, 150.0])
option = LaguerreGaussianOption(p=0, l=1, waist=10.0)
wl = wavelength(340 * m / s)
laguerre_gaussian_phase(geometry, target, np.array([0.0, 0.0, 1.0]), option, wl, phases)
laguerre_gaussian_intensity(geometry, target, np.array([0.0, 0.0, 1.0]), option, wl, intensities)
